### PR TITLE
Replace amrex::Abort with WARPX_ABORT_WITH_MESSAGE

### PR DESCRIPTION
**Pull request by lucafedeli88**
This PR replaces all the occurrences of `amrex::Abort` not in a GPU kernel with the `WARPX_ABORT_WITH_MESSAGE` macro, which standardizes the format and also adds some information on where the issue that triggered the abort message occurred.

### CODE REVIEW

**Comment by ax3l** [comment]
Although the PR does not change any test, this test shows changes in the checksum:
```
2023-06-06T13:05:32.1825672Z [1mworking on test: galilean_rz_psatd[0m
2023-06-06T13:05:32.1825870Z    re-making clean...
2023-06-06T13:05:32.1826009Z    building...
2023-06-06T13:05:32.1826156Z 
2023-06-06T13:05:32.1826399Z [1mconfiguring galilean_rz_psatd build...[0m
2023-06-06T13:05:32.1826555Z    mkdir /tmp/ci/warpx/builddir
2023-06-06T13:05:32.1826940Z    cmake  -DAMReX_ASSERTIONS=ON -DAMReX_TESTING=ON -DWarpX_LIB=ON -DWarpX_DIMS=RZ -DWarpX_PSATD=ON -S /tmp/ci/warpx/ -B /tmp/ci/warpx/builddir 
2023-06-06T13:05:32.1827105Z 
2023-06-06T13:05:32.1827296Z [1mbuilding galilean_rz_psatd...[0m
2023-06-06T13:05:32.1827549Z    cmake --build /tmp/ci/warpx/builddir -j 2 --  
2023-06-06T13:05:32.1827702Z    Compilation time: 25.919 s
2023-06-06T13:05:32.1828000Z    run & test directory: /tmp/ci/rt-WarpX/WarpX-tests/2023-06-06/galilean_rz_psatd/
2023-06-06T13:05:32.1828202Z    copying files to run directory...
2023-06-06T13:13:03.6955943Z    path to input file: Examples/Tests/nci_psatd_stability/inputs_rz
2023-06-06T13:13:03.6956354Z    running the test...
2023-06-06T13:13:03.6957595Z    mpiexec -n 1 ./galilean_rz_psatd.ex inputs_rz  diag1.file_prefix=galilean_rz_psatd_plt   amrex.abort_on_unused_inputs=1 amrex.fpe_trap_invalid=1 amrex.fpe_trap_zero=1 amrex.fpe_trap_overflow=1 warpx.always_warn_immediately=1 warpx.abort_on_warning_threshold=low warpx.do_dynamic_scheduling=0 warpx.serialize_initial_conditions=1 electrons.random_theta=0 ions.random_theta=0 psatd.current_correction=0 warpx.abort_on_warning_threshold=medium
2023-06-06T13:13:03.6958194Z    Execution time: 33.459 s
2023-06-06T13:13:03.6958445Z [33m   WARNING: unable to open the job_info file[0m
2023-06-06T13:13:03.6958610Z    doing the analysis...
2023-06-06T13:13:03.6958829Z [33m   WARNING: analysis failed...[0m
2023-06-06T13:13:03.6959316Z ./analysis_galilean.py  galilean_rz_psatd_plt000400
2023-06-06T13:13:03.6959398Z 
2023-06-06T13:13:03.6959515Z Check numerical stability:
2023-06-06T13:13:03.6959673Z err_energy = 1.862138691139192e-10
2023-06-06T13:13:03.6959806Z tol_energy = 1e-08
2023-06-06T13:13:03.6959988Z ERROR: Benchmark and plotfile checksum have different value for key [lev=0,Ez]
2023-06-06T13:13:03.6960172Z Benchmark: [lev=0,Ez] 4.124588451444761e+03
2023-06-06T13:13:03.6960336Z Plotfile : [lev=0,Ez] 4.124588455632591e+03
2023-06-06T13:13:03.6960624Z Absolute error: 4.19e-06
2023-06-06T13:13:03.6960760Z Relative error: 1.02e-09
2023-06-06T13:13:03.6960834Z 
2023-06-06T13:13:03.6961015Z ----------------
2023-06-06T13:13:03.6961155Z New file for galilean_rz_psatd:
2023-06-06T13:13:03.6961292Z {
2023-06-06T13:13:03.6961398Z   "lev=0": {
2023-06-06T13:13:03.6961540Z     "Bt": 0.0001427751611710155,
2023-06-06T13:13:03.6961679Z     "Er": 42705.92836687861,
2023-06-06T13:13:03.6961833Z     "Et": 198517.8121599192,
2023-06-06T13:13:03.6963011Z     "Ez": 4124.588455632591,
2023-06-06T13:13:03.6963160Z     "divE": 50681.523267918594,
2023-06-06T13:13:03.6963317Z     "jr": 7.07688551916584,
2023-06-06T13:13:03.6963451Z     "jz": 469.93281767722846,
2023-06-06T13:13:03.6963605Z     "rho": 4.488636594605648e-07
2023-06-06T13:13:03.6963725Z   },
2023-06-06T13:13:03.6963976Z   "electrons": {
2023-06-06T13:13:03.6964117Z     "particle_momentum_x": 7.188262495831913e-22,
2023-06-06T13:13:03.6964299Z     "particle_momentum_y": 9.366670273418532e-24,
2023-06-06T13:13:03.6964462Z     "particle_momentum_z": 8.903838605112889e-17,
2023-06-06T13:13:03.6964619Z     "particle_position_x": 633733.1377069331,
2023-06-06T13:13:03.6964789Z     "particle_position_y": 7862152.008302979,
2023-06-06T13:13:03.6964941Z     "particle_theta": 51471.82060096203,
2023-06-06T13:13:03.6965108Z     "particle_weight": 1.0261080645329302e+20
2023-06-06T13:13:03.6965231Z   },
2023-06-06T13:13:03.6965346Z   "ions": {
2023-06-06T13:13:03.6965609Z     "particle_momentum_x": 1.313939911308472e-18,
2023-06-06T13:13:03.6965789Z     "particle_momentum_y": 9.365301692233452e-24,
2023-06-06T13:13:03.6965958Z     "particle_momentum_z": 1.6348806629857118e-13,
2023-06-06T13:13:03.6966136Z     "particle_position_x": 633733.1365353089,
2023-06-06T13:13:03.6966297Z     "particle_position_y": 7862152.007285856,
2023-06-06T13:13:03.6966451Z     "particle_theta": 51471.6099169301,
2023-06-06T13:13:03.6966835Z     "particle_weight": 1.0261080645329302e+20
2023-06-06T13:13:03.6966962Z   }
2023-06-06T13:13:03.6967076Z }
2023-06-06T13:13:03.6967276Z ----------------
2023-06-06T13:13:03.6967334Z 
2023-06-06T13:13:03.6967467Z    Analysis time: 2.202 s
2023-06-06T13:13:03.6967700Z [33m   WARNING: unable to copy analysis image[0m
2023-06-06T13:13:03.6967868Z    archiving the output...
2023-06-06T13:13:03.6968011Z    creating problem test report ...
2023-06-06T13:13:03.6968248Z    [31mgalilean_rz_psatd FAILED[0m
```

**Comment by ax3l** [comment]
@lucafedeli88 can you please slightly bump the tolerance of the `galilean_rz_psatd` test to fit in?

https://github.com/ECP-WarpX/WarpX/issues/2132#issuecomment-1581249307

**Comment by lucafedeli88** [comment]
@ax3l , I've just bumped precision. Now the test passes!

**Review by ax3l — APPROVED** [review]

File: Source/BoundaryConditions/WarpX_PML_kernels.H [hunk 1]
```diff
@@ -30,7 +31,7 @@ void warpx_damp_pml_ex (int i, int j, int k, Array4<Real> const& Ex,
 #if defined(WARPX_DIM_1D_Z)
     amrex::ignore_unused(i, j, k, Ex, Ex_stag, sigma_fac_x, sigma_fac_y, sigma_fac_z,
            sigma_star_fac_x, sigma_star_fac_y, sigma_star_fac_z, xlo, ylo, zlo, dive_cleaning);
-    amrex::Abort("PML not implemented in Cartesian 1D geometry");
+    WARPX_ABORT_WITH_MESSAGE("PML not implemented in Cartesian 1D geometry");
```
Comment (ax3l) [inline Source/BoundaryConditions/WarpX_PML_kernels.H:34]: This is a kernel (host-device function) and thus a bug.
Seen with ROcm 5.2.

File: Source/BoundaryConditions/WarpX_PML_kernels.H [hunk 2]
```diff
@@ -107,7 +108,7 @@ void warpx_damp_pml_ey (int i, int j, int k, Array4<Real> const& Ey,
 #if defined(WARPX_DIM_1D_Z)
     amrex::ignore_unused(i, j, k, Ey, Ey_stag, sigma_fac_x, sigma_fac_y, sigma_fac_z,
            sigma_star_fac_x, sigma_star_fac_y, sigma_star_fac_z, xlo, ylo, zlo, dive_cleaning);
-    amrex::Abort("PML not implemented in Cartesian 1D geometry");
+    WARPX_ABORT_WITH_MESSAGE("PML not implemented in Cartesian 1D geometry");
```
Comment (ax3l) [inline Source/BoundaryConditions/WarpX_PML_kernels.H:111]: host-device function

Comment (ax3l) [reply Source/BoundaryConditions/WarpX_PML_kernels.H:111]: #3987

File: Source/BoundaryConditions/WarpX_PML_kernels.H (hunk 1 above)
Comment (ax3l) [reply Source/BoundaryConditions/WarpX_PML_kernels.H:34]: #3987

**Pull request merged**